### PR TITLE
Dive into Batch scripting language

### LINKED ISSUE #599: Dive into Batch scripting language
### Brief Summary: 

**Batch** is a native programming language for windows, it can execute certain commands automatically at the level of an operating system ", it can be classified as an interpreter-based scripting language.
Also, it can be used for server installations.
This tutorial will be a "dive into" from the basics to the advanced points.

### Key Takeaways: 

- What is **Batch** programming 
- **Batch** variables/loops/ etc..
- How to manipulate files in the system (copy, move, delete, etc..)
- I will create 2 or 3 advanced apps to take the reader from the beginner side to the advanced one.

### References: 
N/A

### PR DESCRIPTION
**Pull request by ahmadmardeni1**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #599

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
this fixes #599

**Comment by hectorkambow** [comment]
plagiarism check done

File: articles/dive-into-batch-scripting language/index.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+Before we dive right in, the reader would need to have the following:
+- A good understanding of any programming language.
+- A basic understanding of system programming.
+
+### Create a Batch file
+Creating a **Batch** file is very easy, you have to open any text editor then saving that file with an extension **.bat**.
+If you want to run that file you need to click on it or calling it by the command prompt.
+
+### Let's Code!
+By default, when you run a **Batch** file it displays every command is currently executing, in order to turn off this display you have to use:
```
Comment (botahamec) [inline articles/dive-into-batch-scripting language/index.md:21]: This should be two sentences

File: articles/dive-into-batch-scripting language/index.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
+# Dive-into-Batch-scripting-language
+### Introduction
+**Batch** commands are often used to start programs or to control them in the Windows environment. Also, it can be used for server installations.
+If you write a sequence of **Batch** commands and save the file with the extension **.bat** or **.cmd** then the file is called **Batch file**.
```
[the remark this answers is not in the document]
Comment (botahamec) [reply articles/dive-into-batch-scripting language/index.md:4]: This last sentence can be shortened significantly. In fact, many of your sentences can. Did you use Hemingway to check your writing?

File: articles/dive-into-batch-scripting language/index.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+```
+In the code above you can make a backup of the "first input folder" %1 files and its subfolders files to the second folder %2.
+To include any hidden file you have to add the /h command-line option also you can add /e to include any empty subdirectories.
+
+Now let's create a simple code to delete all temporary files with the extension **TMP**:
+```batch
+@echoo off
+del %1\*.tmp
+```
+The execution process of a **Batch** file is line-by-line. However, you can use "goto" command which gives you the capability to jump into a particular section of your code.
```
Comment (botahamec) [inline articles/dive-into-batch-scripting language/index.md:69]: **the** "goto" command

File: articles/dive-into-batch-scripting language/index.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
+# Dive-into-Batch-scripting-language
+### Introduction
+**Batch** commands are often used to start programs or to control them in the Windows environment. Also, it can be used for server installations.
+If you write a sequence of **Batch** commands and save the file with the extension **.bat** or **.cmd** then the file is called **Batch file**.
+
+#### Why Batch programming language?
+- **Powerful**: you can execute a lot of complicated operations by one command.
+- The commands are **short** so that they are easy to remember.
+- Major **time saving**.
```
[the remark this answers is not in the document]
Comment (botahamec) [reply articles/dive-into-batch-scripting language/index.md:9]: It should also be "savings" not "saving"

File: articles/dive-into-batch-scripting language/index.md [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+### Create a Batch file
+Creating a **Batch** file is very easy, you have to open any text editor then saving that file with an extension **.bat**.
+If you want to run that file you need to click on it or calling it by the command prompt.
+
+### Let's Code!
+By default, when you run a **Batch** file it displays every command is currently executing, in order to turn off this display you have to use:
+```batch
+@echoo off
+```
+We used "@" sign to made the command apply to itself too.
```
[the remark this answers is not in the document]
Comment (botahamec) [reply articles/dive-into-batch-scripting language/index.md:25]: It should be "**the** @ sign"

**Review by botahamec — CHANGES_REQUESTED** [review]
General comments:

* This was an accurate introduction Batch
* The spelling and grammar need work. It's not as bad as some articles I've reviewed, but still not very good.
* Some of the examples aren't very useful, and will be hard for the reader to actually run. Please change those
* Some parts of the article need further explanation, because they aren't very clear.
* I'm not sure if you ran this through Hemmingway. There are multiple things that Hemmingway highlights that I can see being improved. Please check it again.

**Comment by ahmadmardeni1** [comment]
Hello @botahamec 
Thank you for your suggestions, I did the changes.
The tutorial I submitted was an old one(before I used Grammarly), sorry, my bad.
A few things I want to mention:
- The code you used will not work because it didn't work with the env variable, you can try :
```
@echo off
set ahmad123
If %ERRORLEVEL% == 1 (echo error)
```
It will display "error" because the variable name is not found.

- I didn't explain "if" because it is very clear for any programmer ( one of the Prerequisites was "A good understanding of any programming language")
Thank you again and I will take this suggestion to my next tutorials as well.

**Comment by botahamec** [comment]
@ahmadmardeni1 You should put that code you just gave me in the article, because the reader needs to know that

**Comment by ahmadmardeni1** [comment]
Done @botahamec

**Pull request merged**